LaTeX/TeX.sublime-syntax

```diff
@@ -277,30 +272,51 @@ contexts:
     - include: paragraph-pop
 
   def-definition:
-    - match: \{
-      scope: punctuation.definition.group.brace.begin.tex
-      set: def-definition-body
+    - match: (?=\{)
+      set:
+        - def-scope-as-body
+        - macro-braces-begin
     - include: paragraph-pop
     - include: else-pop
 
-  def-definition-body:
-    - meta_scope: meta.function.body.tex meta.group.brace.tex
-    - include: macro-brace-group-body
+  def-scope-as-body:
+    - meta_scope: meta.function.body.tex
+    - include: immediately-pop
 
-  # within macros, it is possible that only part of some nested struture
-  # is present. To prevent this from causing problems, here we only match
-  # elements that are simple, i.e. that do not push to the stack.
   macro-braces:
+    - match: (?=\{)
+      push: macro-braces-begin
+
+  macro-braces-begin:
     - match: \{
       scope: punctuation.definition.group.brace.begin.tex
-      push: macro-brace-group-body
+      set: macro-braces-content
+
+  macro-braces-end:
+     - match: \}
+       scope: punctuation.definition.group.brace.end.tex
+       pop: 1
 
-  macro-brace-group-body:
+  # within macros, it is possible that only part of some nested struture
+  # is present. Don't include any context's here that break if they are
+  # only partially matched.
+  macro-braces-content:
     - meta_scope: meta.group.brace.tex
-    - include: brace-group-end
+    - include: macro-braces
+    - include: macro-braces-end
```

Comment:
With the former suggestion `macro-braces-end` and `meta_scope` become obsolete in this context.

```suggestion
```